.github/workflows/tests.yml

```diff
@@ -40,7 +40,7 @@ jobs:
         bundler-cache: true # runs 'bundle install' and caches installed gems automatically (not working?)
     - uses: ./.github/actions/setup-test-environment
     - name: Install JavaScript dependencies with Yarn
-      run: yarn check || yarn install;
+      run: yarn check || yarn install --frozen-lockfile;
```

Comment:
I think we do not add modules so frequently, right? Can we cache node_modules folder somehow and do `yarn install` only if package.json or yarn.lock changes to speed things up?

Reply:
I looked this up and found this was the most common and "standard" way to speed up yarn installs, but yeah, i think we can use one of these:

https://dev.to/mpocock1/how-to-cache-nodemodules-in-github-actions-with-yarn-24eh

https://github.com/actions/cache/blob/main/examples.md#node---yarn

Let's try in a different PR? Thanks!